### ISSUE TITLE
Open IB BTL error spam (retry_exceeded_error)

### ISSUE THREAD

**Issue by abouteiller** [ulfm-devel/ompi#16]
**[Original report](https://bitbucket.org/icldistcomp/ulfm2/issue/16) by Aurelien Bouteiller (Bitbucket: [abouteiller](https://bitbucket.org/abouteiller), GitHub: [abouteiller](https://github.com/abouteiller)).**

----------------------------------------

```
[[43897,1],10][../../../../../trunk/ompi/mca/btl/openib/btl_openib_component.c:3504:handle_wc] from d /home/bouteill/ompi/FTMPI3/RTS-ULFM/run/share/openmpi/amca-param-sets:/home/bouteill/ompi/FTMPI3/ul│ancer01 to: dancer03 error polling LP CQ with status RETRY EXCEEDED ERROR status number 12 for wr_id -------------------------------------------------------------------------- │13dec80 opcode 128 vendor error 129 qp_idx 0 
-------------------------------------------------------------------------- │[[43897,1],10][../../../../../trunk/ompi/mca/btl/openib/btl_openib_component.c:3504:handle_wc] from d Process 23584 Unable to locate the parameter file "enable-ft-mpi" in the following search path: │ancer01 to: dancer03 error polling HP CQ with status WORK REQUEST FLUSHED ERROR status number 5 for w /home/bouteill/ompi/FTMPI3/RTS-ULFM/run/share/openmpi/amca-param-sets:/home/bouteill/ompi/FTMPI3/ul│r_id 1663300 opcode 128 vendor error 244 qp_idx 0
```

**Comment by abouteiller** [comment]
**Original comment by Aurelien Bouteiller (Bitbucket: [abouteiller](https://bitbucket.org/abouteiller), GitHub: [abouteiller](https://github.com/abouteiller)).**

----------------------------------------

This is not a harmless error message. While this "retry" activity is ongoing progress stalls.

**Comment by abouteiller** [comment]
**Original comment by Aurelien Bouteiller (Bitbucket: [abouteiller](https://bitbucket.org/abouteiller), GitHub: [abouteiller](https://github.com/abouteiller)).**

----------------------------------------

```
too many retries sending message to 0x0003:0x00d00412, giving up
```

**Comment by abouteiller** [comment]
**Original comment by Aurelien Bouteiller (Bitbucket: [abouteiller](https://bitbucket.org/abouteiller), GitHub: [abouteiller](https://github.com/abouteiller)).**

----------------------------------------

Progress stalls completely after such reports arrive. Needs to be fixed.

**Comment by abouteiller** [comment]
**Original comment by Aurelien Bouteiller (Bitbucket: [abouteiller](https://bitbucket.org/abouteiller), GitHub: [abouteiller](https://github.com/abouteiller)).**

----------------------------------------

Principal issue resolved in 04f61d22 and 79aca0bb : credits were not returned from communication with failed procs. 

Issue remains for UDCM failures (and presumably for rdmacm as well).

**Comment by abouteiller** [comment]
**Original comment by Aurelien Bouteiller (Bitbucket: [abouteiller](https://bitbucket.org/abouteiller), GitHub: [abouteiller](https://github.com/abouteiller)).**

----------------------------------------

UDCM issue resolved in 2fb5440a
RDMACM does not appear to have a similar code path, the issue could not be exhibited on our EDR hardware.
